XMonad/Actions/TreeSelect.hs

```diff
@@ -535,6 +537,22 @@ navigate = gets tss_display >>= \d -> join . liftIO . allocaXEvent $ \e -> do
             fromMaybe navigate $ M.lookup (mask, fromMaybe xK_VoidSymbol ks) f
       else return navigate
 
+{- Note [Allow Events]
+
+First of all, read "XMonad.Prompt Note [Allow ButtonEvents]" to
+understand what we are doing and why we are doing it.
+
+We are doing something a little different here: because this module does
+not have an analog to 'XMonad.Prompt.handleOther', we instead always
```

Comment:
Well, you could turn
```
if ev_event_type ev == keyPress
```
into
```
case ev_event_type ev of
    t | t == keyPress -> …
    t | t == buttonPress -> …
```

Or just paste the entire
```
    when (t == buttonPress) $ do
        d <- gets dpy
        io $ allowEvents d replayPointer currentTime
```
in here.

---

It's probably overly nitpicky to call for consistency in these matters, but since you already bothered to add several paragraph of comments to justify what you did, it might just be easier to do it consistenly instead and drop the extra text. ;-)

Reply:
Nono, you are totally right of course!  I didn't think about TreeSelect very much, but it definitely is good to be consistent across modules.

N.b.: we should probably (not in this pr, just in general) also apply the `bracket` pattern to TreeSelect, as was done in 09a5fa111c0fc02fa0c49ac9052ac9ce4fec0110